packages/http/package.json

```diff
@@ -16,7 +16,7 @@
     "node": ">=8"
   },
   "dependencies": {
-    "@stoplight/prism-core": "0.0.0",
+    "@stoplight/prism-core": "3.0.0-pre",
```

Comment:
Hmm, curious how this will work.
Let's say you publish this, am I right that:

`lerna` will look at the remote tags, will find that latest tag is `2.x.y` and will bump to `2.x+1.y`?

OR

will `lerna` look at the `package.json#version` and bump to `3.0.0-pre.1` (or similar) then tag and release `3.0.0-pre.1`? But at the same time `@stoplight/prism-core: 3.0.0-pre` will still remain unpublished thus this code will not build?

Reply:
Good question. Let's break down the current publish command

`yarn lerna publish prerelease --yes --dist-tag beta --conventional-commits`

* `prerelease`: add the `-alpha` to the final version
* `--yes`: do not ask for any confirmation
* `--dist-tag beta`: when publish on NPM, put it on the beta tag (I should change it to alpha).
* `--conventional-commits`: use angular style commits to determine the next version (the same we're using in semantic-release)


Let's make an example of what's going to happen right now:

lerna will parse the current version in package.json (3.0.0-pre -> 3.0.0), realise that it's a prerelease and so add the `-alpha` tag and then make a minor bump. Final result: `3.1.0-alpha`

Since the `dist-tag` is `beta`, the package will be installable only when specifying it explicitly

`npm i @stoplight/prism-cli`  -> No result
`npm i @stoplight/prism-cli@beta` -> OK

This will make sure that nobody installs and uses the software until we feel comfortable.

Does that clarify?

Reply:
That helps a lot, still have some questions :)

> lerna will parse the current version in package.json (3.0.0-pre -> 3.0.0), realise that it's a prerelease and so add the -alpha tag and then make a minor bump. Final result: 3.1.0-alpha

- Will lerna update package.json#version? I think workspaces rely on that value for local development, right?
- Why even `3.1.0`? Shouldn't we at least release `3.0.0` first? Isn't lerna simply adding `-alpha` w/o bumping minor if it notices it's a prerelease?

> Since the dist-tag is beta, the package will be installable only when specifying it explicitly

And when would a `stable` release be triggered? Do you plan to add it too?

> npm i @stoplight/prism-cli@beta -> OK

Still, even though `@stoplight/prism-core` is now released as `3.1.0-alpha`, @stoplight/prism-http` points to `@stoplight/prism-core@3.0.0-pre` right? If I `yarn add @stoplight/prism-http` it won't be able to find `@stoplight/prism-core@3.0.0-pre` and will break.

Reply:
> Will lerna update package.json#version?

Yes it does for the package itself and for all the dependants packages it's indeed required and we can't escape that.

> Isn't lerna simply adding -alpha w/o bumping minor if it notices it's a prerelease?

You're actually right, I wrote the version incorrectly in the comment

<img width="589" alt="image" src="https://user-images.githubusercontent.com/1416224/55869337-9b413d00-5b86-11e9-8fcc-238a7e6a216b.png">


> And when would a `stable` release be triggered? Do you plan to add it too?

The default tag is `latest`, and that's the one we're going to release when the stuff gets in `master`, but this is definitely something that'll be coming next.

> Still, even though @stoplight/prism-core is now released as 3.1.0-alpha, @stoplight/prism-http points to @stoplight/prism-core@3.0.0-pre right? If I yarn add @stoplight/prism-http@beta it won't be able to find @stoplight/prism-core@3.0.0-pre and will break.

I'm not too sure I'm following here; all the versions are updated in the relative package.json files before they are published, no mismatch should be happening.

Moreover, you can test the changes locally on your computer before publishing by doing 

`yarn lerna publish prerelease --dist-tag beta --conventional-commits`

notice the missing yes parameter:

```diff
- yarn lerna publish prerelease --yes --dist-tag beta --conventional-commits
+ yarn lerna publish prerelease --dist-tag beta --conventional-commits
```

and then answering no — you should be able to have a picture of what's going on.

Reply:
Ok, all sounds good!